k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapResponseParser.java

```diff
@@ -109,7 +109,7 @@ private void readTaggedResponse(ImapResponseCallback callback) throws IOExceptio
                 continue;
             }
 
-            if (untaggedHandler != null) {
+            if (response.getTag() == null && untaggedHandler != null) {
```

Comment:
I think it only didn't fail a test because the test was badly written. I wrote some better tests for it and I think it should have failed. To me it looked like faulty logic and I browsed the spec a bit.

I'll keep using it on my device though.